libc/src/__support/float_to_string.h

```diff
@@ -548,7 +550,8 @@ class FloatToString {
 
       val = POW10_SPLIT_2[p];
 #endif
-      const int32_t shift_amount = SHIFT_CONST + (-exponent - IDX_SIZE * idx);
+      const int32_t shift_amount =
+          static_cast<int32_t>(SHIFT_CONST + (-exponent - IDX_SIZE * idx));
```

Comment:
Sorry, I think I merged this in https://github.com/llvm/llvm-project/pull/74369. I wonder what page I was looking at where I didn't observe warnings on the above?

Reply:
I've rebased to pick that up, and also matched the narrower cast scope you had.